### PR TITLE
[plugin.video.catchuptvandmore] 0.2.24

### PR DESCRIPTION
**Pull request by sy6sy2**
### Description

[Version 0.2.24]

[Fix/DRM] Live TV - RTBF (LaUne, LaDeux, ...) (BE)
[Fix] Live TV - Télé Quebec (CA)
[Remove] Live TV - Yoopa and LCN (CA) - No present in the website of TVA.ca
[Rename] Live TV - NPO Zapp to NPO 3 (NL) - Need more work
[Fix] Live TV - BFM Business (FR)
[Fix] Website - Allocine (FR)
[Move] Catch-Up TV - France TV (FR) - All France TV channels are now in 'france.tv' folder like the official website
[Add] Live TV - Equidia (FR)
[Fix] Website - Têtes à claques (CA) - Add Zarbi videos
[Fix] Live TV - Nessan (TN)
[Fix] Catch-Up TV - Guilli (FR)
[Add] Live TV - B Smart (FR)
[Fix] Catch-up TV - NYC Media (US)
[Fix] Cathc-up TV - Arte Concert (FR)
[Move] Catch-Up TV - France TV (FR) - FR3 régions and 1ère outre-mer are now in 'france.tv' folder

### Checklist:

- [X] My code follows the [add-on rules](http://kodi.wiki/view/Add-on_rules) and [piracy stance](http://kodi.wiki/view/Official:Forum_rules#Piracy_Policy) of this project. 
- [X] I have read the [CONTRIBUTING](https://github.com/xbmc/repo-plugins/blob/master/CONTRIBUTING.md) document
- [X] Each add-on submission should be a single commit with using the following style: [plugin.video.foo] v1.0.0

Additional information :
- Submitting your add-on to this specific branch makes it available to any Kodi version equal or higher than the branch name with the applicable Kodi dependencies limits.
- [add-on development](http://kodi.wiki/view/Add-on_development) wiki page.
- Kodi [pydocs](http://kodi.wiki/view/PyDocs) provide information about the Python API
- [PEP8](https://www.python.org/dev/peps/pep-0008/) codingstyle which is considered best practice but not mandatory.
- This add-on repository has automated code guideline check which could help you improve your coding. You can find the results of these check at [Codacy](https://www.codacy.com/app/Kodi/repo-plugins/dashboard). You can create your own account as well to continuously monitor your python coding before submitting to repo.
- Development questions can be asked in the [add-on development](http://forum.kodi.tv/forumdisplay.php?fid=26) section on the Kodi forum.
- If you see no activity on your PR after a week (so at least one weekend has passed) then please go to the #kodi-dev freenode IRC channel to reach out to the team

### CODE REVIEW

File: plugin.video.catchuptvandmore/resources/lib/channels/fr/bsmart.py [hunk 1]
```diff
@@ -0,0 +1,60 @@
+# -*- coding: utf-8 -*-
+"""
```
Comment (enen92) [inline plugin.video.catchuptvandmore/resources/lib/channels/fr/bsmart.py:2]: would be nice if at some point you can drop such large headers and adopt the spdx ones. Have a look into the kodi source code for an example

File: plugin.video.catchuptvandmore/resources/lib/channels/fr/francetv.py [hunk 2]
```diff
@@ -32,390 +32,373 @@
...
-    Seems to be broken in current CodeQuick version
-    yield Listitem.recent(
-        list_videos_last,
-        item_id = item_id)
-    """
+    # Categories
     item = Listitem()
-    item.label = "Recent"
-    item.set_callback(list_videos_last, item_id=item_id)
+    item.label = 'Catégories'
```
Comment (enen92) [inline plugin.video.catchuptvandmore/resources/lib/channels/fr/francetv.py]: is this shown in the interface? If so it needs localization

File: plugin.video.catchuptvandmore/resources/lib/channels/fr/francetv.py [hunk 3]
```diff
@@ -32,390 +32,373 @@
...
 
-            plot += show['text']
 
-            last_publication_date = show['dates']['last_publication_date']
-            publication_date = time.strftime(
-                "%Y-%m-%d", time.localtime(last_publication_date))
+@Route.register
+def outre_mer_root(plugin, region_path):
+    menu_items = [
+        ('Les dernières vidéos', '/generic/taxonomy/%s/contents'),
```
Comment (enen92) [inline plugin.video.catchuptvandmore/resources/lib/channels/fr/francetv.py]: the same for these

File: plugin.video.catchuptvandmore/resources/lib/channels/fr/francetv.py [hunk 4]
```diff
@@ -32,390 +32,373 @@
...
-                              item_id=item_id,
-                              id_yatta=show['id'])
-            item_post_treatment(item, is_playable=True, is_downloadable=True)
-            yield item
-        page = page + 1
+            if set_item_callback_based_on_type(item, collection['type'], collection, next_page_item):
+                items.append(item)
+    if 'item' in j and 'program_path' in j['item']:
+        item = Listitem()
+        item.label = 'Toutes les vidéos'
```
Comment (enen92) [inline plugin.video.catchuptvandmore/resources/lib/channels/fr/francetv.py]: same here

File: plugin.video.catchuptvandmore/resources/lib/channels/fr/bsmart.py (hunk 1 above)
Comment (sy6sy2) [reply plugin.video.catchuptvandmore/resources/lib/channels/fr/bsmart.py:2]: I will look at that for the next release, thank you.

**Review by enen92 — COMMENTED** [review]
Please comment on my comments (except the spdx thing), if they are false positives this is good to merge.

File: plugin.video.catchuptvandmore/resources/lib/channels/fr/francetv.py (hunk 2 above)
Comment (sy6sy2) [reply plugin.video.catchuptvandmore/resources/lib/channels/fr/francetv.py]: Yes it's only shown in the French TV section.
Does it needs a localization even if it's only in the French "section" of the add-on?

Comment (sy6sy2) [reply plugin.video.catchuptvandmore/resources/lib/channels/fr/francetv.py]: Should be ok now 😉

**Review by enen92 — APPROVED** [review]

**Pull request merged**